File: SDKs/github.com/ghes-2.22/1.1.4/typescript/src/sdk/models/operations/gitgettag.ts
import { Metadata, SpeakeasyBase } from "../../../internal/utils/utils";
import * as shared from "../shared";


export class GitGetTagPathParams extends SpeakeasyBase {
  @Metadata({ data: "pathParam, style=simple;explode=false;name=owner" })
  owner: string;

  @Metadata({ data: "pathParam, style=simple;explode=false;name=repo" })
  repo: string;

  @Metadata({ data: "pathParam, style=simple;explode=false;name=tag_sha" })
  tagSha: string;
}


export class GitGetTagRequest extends SpeakeasyBase {
  @Metadata()
  pathParams: GitGetTagPathParams;
}


export class GitGetTagResponse extends SpeakeasyBase {
  @Metadata()
  contentType: string;

  @Metadata()
  statusCode: number;

  @Metadata()
  basicError?: shared.BasicError;

  @Metadata()
  gitTag?: shared.GitTag;
}

File: SDKs/github.com/ghes-3.1/1.1.4/python/src/sdk/models/operations/git_get_tag.py
from dataclasses import dataclass, field
from typing import Optional
from sdk.models import shared


@dataclass
class GitGetTagPathParams:
    owner: str = field(metadata={'path_param': { 'field_name': 'owner', 'style': 'simple', 'explode': False }})
    repo: str = field(metadata={'path_param': { 'field_name': 'repo', 'style': 'simple', 'explode': False }})
    tag_sha: str = field(metadata={'path_param': { 'field_name': 'tag_sha', 'style': 'simple', 'explode': False }})
    

@dataclass
class GitGetTagRequest:
    path_params: GitGetTagPathParams = field()
    

@dataclass
class GitGetTagResponse:
    content_type: str = field()
    status_code: int = field()
    basic_error: Optional[shared.BasicError] = field(default=None)
    git_tag: Optional[shared.GitTag] = field(default=None)
    
